### DISCUSSION TITLE
Saving new objects in Links (3.0)

### DISCUSSION THREAD

**Discussion by etiennelenhart** [isar/isar#477] [Help]
Now that `IsarLinks` doesn't save new objects anymore, what's the correct way to do this in Isar 3.0?

Iterating on the `Student` & `Teacher` example from the docs, when would I save the newly created `Teacher`?
```dart
@Collection()
class Teacher {
  int? id;
  late String subject;
}

@Collection()
class Student {
  int? id;
  late String name;
  final teachers = IsarLinks<Teacher>();
}

Future<void> save(Isar isar) async {
  final mathTeacher = Teacher()..subject = 'Math';

  final linda = Student()
    ..name = 'Linda'
    ..teachers.add(mathTeacher);

  await isar.writeTxn(() async {
    await linda.teachers.save(); // 1
    await isar.students.put(linda);
    await linda.teachers.save(); // 2
  });
}
```
`1` fails with "LateInitializationError: Field 'targetCollection' has not been initialized."
`2` fails with "Object Instance of 'Teacher' has no id and can therefore not be linked."

**Comment by swood36584** [comment]
Looks like you are not saving the id when saving the student.
Change the save to this:
`linda.id = await isar.students.put(linda);`

**Reply by etiennelenhart** [reply]
Sorry, I clarified the error message. `2` complains about `Teacher` missing an `id`. Which makes sense, but I'd expect `IsarLinks` to put the `Teacher` in its collection first, before linking it to the `Student`.
My main question would be, whether I have to manually do this now in 3.0, although that would be kind of a bummer.
